### PR TITLE
[Eager] Support SellectedRows MergeAdd case

### PR DESCRIPTION
**Pull request by veyron95**
### PR types
Function optimization

### PR changes
Others 

### Describe
1. Refactor SelectedRowsMerge func by using template
2. Add related unit test in grad_tensor_holder_test.cc and test_gradient_accumulator.cc

### CODE REVIEW

File: paddle/fluid/imperative/gradient_accumulator.cc [hunk 1]
```diff
@@ -469,23 +480,25 @@ template void SelectedRowsAddTensor(
...
   platform::DeviceContextPool& pool = platform::DeviceContextPool::Instance();
 
   std::vector<const pten::SelectedRows*> src_selected_rows;
   src_selected_rows.emplace_back(&src_selected_rows1);
   src_selected_rows.emplace_back(&src_selected_rows2);
-  auto dst_var = std::make_shared<VariableWrapper>("Temp");
-  auto* dst_selected_rows =
-      dst_var->MutableVar()->GetMutable<pten::SelectedRows>();
+
+  auto dst_var = std::make_shared<ReturnVarType>("Temp");
```
Comment (JiabinYang) [inline paddle/fluid/imperative/gradient_accumulator.cc:505]: Conceal this instead of modify GetInnerMutableTensor

**Review by JiabinYang — COMMENTED** [review]
some comments

**Comment by chenbinqun** [comment]
邮件已收到！！

File: paddle/fluid/imperative/gradient_accumulator.cc [hunk 2]
```diff
@@ -247,12 +240,29 @@ TType& GetInnerTensor(const paddle::experimental::Tensor& src) {
...
       platform::errors::Fatal("We only add tensor with value if a tensor is "
-                              "NOT INITILIZED, it should just move instead of "
+                              "INITILIZED, it should just move instead of "
                               "calling this method."));
   auto* src_tensor = static_cast<TType*>(src.impl().get());
   return *src_tensor;
 }
 
+template <typename TType>
+TType* GetInnerMutable(paddle::experimental::Tensor* dst) {
```
Comment (JiabinYang) [inline paddle/fluid/imperative/gradient_accumulator.cc]: Bad name

Comment (JiabinYang) [reply paddle/fluid/imperative/gradient_accumulator.cc]: GetEmptyInnerTensor

**Review by JiabinYang — COMMENTED** [review]
some comments

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**